File: app/src/main/java/igrek/todotree/domain/treeitem/CheckboxTreeItem.java
package igrek.todotree.domain.treeitem;


public class CheckboxTreeItem extends AbstractTreeItem {
	
	private String name;
	private boolean checked;
	
	public CheckboxTreeItem(AbstractTreeItem parent, String name, boolean checked) {
		super(parent);
		this.name = name;
		this.checked = checked;
	}
	
	@Override
	public CheckboxTreeItem clone() {
		return new CheckboxTreeItem(null, name, checked).copyChildren(this);
	}
	
	@Override
	public String getDisplayName() {
		return name;
	}
	
	@Override
	public String getTypeName() {
		return "checkbox";
	}
	
	public void setName(String name) {
		this.name = name;
	}
	
	public boolean isChecked() {
		return checked;
	}
	
	public void setChecked(boolean checked) {
		this.checked = checked;
	}
}

File: app/src/main/java/igrek/todotree/domain/treeitem/CheckboxTreeItem.kt
package igrek.todotree.domain.treeitem

class CheckboxTreeItem(parent: AbstractTreeItem?, override var displayName: String, var isChecked: Boolean) : AbstractTreeItem(parent) {
    override fun clone(): CheckboxTreeItem {
        return CheckboxTreeItem(null, displayName, isChecked).copyChildren(this)
    }

    override val typeName: String
        get() = "checkbox"

    fun setName(name: String) {
        displayName = name
    }

}